### PR TITLE
Tests: Unmap memory ranges on /dev/mem after testing mmap(2)

### PR DESCRIPTION
**Pull request by supercomputer7**
cc @IdanHo

### CODE REVIEW

File: Tests/Kernel/TestMemoryDeviceMmap.cpp [hunk 1]
```diff
@@ -23,7 +23,11 @@ static ALWAYS_INLINE bool mem_chunk(int fd, u64 base, u64 length)
 {
     u64 mmoffset = base % sysconf(_SC_PAGESIZE);
     void* mmp = mmap(NULL, mmoffset + length, PROT_READ, MAP_SHARED, fd, base - mmoffset);
-    return mmp != MAP_FAILED;
+    bool success = (mmp != MAP_FAILED);
+    if (munmap(mmp, mmoffset + length) < 0) {
```
Comment (IdanHo) [inline Tests/Kernel/TestMemoryDeviceMmap.cpp:27]: We probably shouldn't try to munmap if the mmap failed.

**Review by IdanHo — COMMENTED** [review]
no need for the `done` in the commit title

File: Tests/Kernel/TestMemoryDeviceMmap.cpp (hunk 1 above)
Comment (supercomputer7) [reply Tests/Kernel/TestMemoryDeviceMmap.cpp:27]: Right. I'll fix it right away :)

File: Tests/Kernel/TestMemoryDeviceMmap.cpp [hunk 2]
```diff
@@ -23,7 +23,12 @@ static ALWAYS_INLINE bool mem_chunk(int fd, u64 base, u64 length)
 {
     u64 mmoffset = base % sysconf(_SC_PAGESIZE);
     void* mmp = mmap(NULL, mmoffset + length, PROT_READ, MAP_SHARED, fd, base - mmoffset);
-    return mmp != MAP_FAILED;
+    if (mmp == MAP_FAILED)
+        return false;
+    if (munmap(mmp, mmoffset + length) < 0) {
```
Comment (IdanHo) [inline Tests/Kernel/TestMemoryDeviceMmap.cpp:28]: Sorry, just one last nitpick, no need for the curly braces `{}`

Comment (supercomputer7) [reply Tests/Kernel/TestMemoryDeviceMmap.cpp:28]: I'll delete these braces but personally I don't think there's a problem with these.

Comment (IdanHo) [reply Tests/Kernel/TestMemoryDeviceMmap.cpp]: It's not a functional change, just a stylistic one, but it is the dominant style of the project, so let's be consistent about it.

**Review by IdanHo — APPROVED** [review]
Thanks!

**Pull request merged**